vms/platformvm/validators/manager.go

```diff
@@ -170,56 +171,105 @@ func (m *manager) GetValidatorSet(ctx context.Context, height uint64, subnetID i
 		return nil, ErrMissingValidator
 	}
 
-	currentSubnetValidatorList := currentSubnetValidators.List()
-	vdrSet := make(map[ids.NodeID]*validators.GetValidatorOutput, len(currentSubnetValidatorList))
-	for _, vdr := range currentSubnetValidatorList {
-		primaryVdr, ok := currentPrimaryNetworkValidators.Get(vdr.NodeID)
+	currentSubnetValidatorList := currentSubnetValidatorSet.List()
+	currentSubnetValidators := make([]*maskedValidator, 0, len(currentSubnetValidatorList))
+	currentSubnetValidatorIndices := make(map[ids.NodeID]int)
+	for i, validator := range currentSubnetValidatorList {
+		primaryVdr, ok := currentPrimaryNetworkValidators.Get(validator.NodeID)
```

Comment:
So annoying to me that we pull up the primary network validator here. It's prolly best to avoid duplicating the BLS key across stakers of different subnets. I wonder how much memory it'd cost to duplicate the bls key across stakers and avoid calling primary network validator here.